domManipulation.js

```diff
@@ -0,0 +1,157 @@
+import { format } from "date-fns";
+
+// Get the current date to convert the forecast for the current date to "Today";
+export function getCurrentDate() {
+  // Convert to format that matches how the result from the network call has been formulated.
+  return format(new Date(), "MMdd");
+}
+
+// Convert the general weather types to corresponding emojis.
+export function weatherIconConverter(weatherType = "") {
+  let weatherEmoji = "";
+  if (weatherType.includes("clearday")) {
+    weatherEmoji = "&#x1F31E;";
+  } else if (weatherType.includes("clearnight")) {
+    weatherEmoji = "&#x1F319";
+  } else if (weatherType.includes("clear")) {
+    weatherEmoji = "&#x1F31E;";
+  } else if (weatherType.includes("cloudy")) {
+    weatherEmoji = "&#x26C5;";
+  } else if (weatherType.includes("shower")) {
+    weatherEmoji = "&#x1F326;";
+  } else if (weatherType.includes("humid")) {
+    weatherEmoji = "&#x1F31E;";
+  } else if (weatherType.includes("rain")) {
+    weatherEmoji = "&#x1F326;";
+  } else if (weatherType.includes("snow")) {
+    weatherEmoji = "&#x2744;";
+  } else if (weatherType.includes("rainsnow")) {
+    weatherEmoji = "&#x1F328;";
+  } else if (weatherType.includes("ts")) {
+    weatherEmoji = "&#x1F329;";
+  }
+
+  return weatherEmoji;
+}
+
+function temperatureColourConverter(temp) {
+  switch (true) {
+    case temp < 0:
+      return "#1818d7";
+    case temp < 10:
+      return "#0e7ccb";
+    case temp < 20:
+      return "#03a3d9";
+    case temp < 30:
+      return "#338a5a";
+    case temp < 40:
+      return "#d38e0e";
+    case temp < 50:
+      return "#ff0000";
+    default:
+      return "#6153cc";
+  }
+}
+
+export function loadWeatherForecast(location, fetchResults) {
+  // Showing the loader for the popup first.
+
+  document.getElementById("popup-loading-container").style.visibility = "visible";
```

Comment:
You may want to handle this separately as you have with hiding the loading container. 
This is going to only start the loading visibility _after_ the results already have come back.

Reply:
This loader is just used for the pop-up, but I have moved it into a new function called `displayForecastLoader` function and called it before any fetch is done to prevent the loader from displaying too long in the popup.